src/react/Orb.js

```diff
@@ -0,0 +1,305 @@
+﻿"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3","../common/HTMLWidget","../common/Utility","../common/PropertyExt","css!orb", "css!./Orb"], factory);
+    } else {
+        root.template_Orb = factory(root.d3, root.common_HTMLWidget, root.common_Utility, root.common_PropertyExt, root.React);
+    }
+}(this, function (d3, HTMLWidget, Utility, PropertyExt){
+    var orb = null;
+    function Orb(target) {
+        HTMLWidget.call(this);
+        this.orbFields =[];
+        this.savedField = [];
+        this.rowFields = [];
+        this.dataFields = [];
+        this.columnFields = [];
+
+    }
+
+
+function mapping(owner){
+ PropertyExt.call(this);
+ this._owner = owner;
+}
+
+mapping.prototype = Object.create(PropertyExt.prototype);
+mapping.prototype.constructor = mapping;
+mapping.prototype._class += " react_Orb";
+mapping.prototype.publish("addField", "", "set", "Show Toolbox or not",function() { return this._owner ? this._owner.columns() : [];}, {optional: true} );
+mapping.prototype.publish("location", true, "set", "Data Location",['row','column','data'], { tags: ["basic"] });
+mapping.prototype.publish("aggregateFunc", "", "set", "Aggregate Function type",['sum','count','min','max','avg','prod','var','varp','stdev','stdevp'], {optional: true} );
+mapping.prototype.publish("formatFunction","","string","Format function");
+
+
+
+Orb.prototype = Object.create(HTMLWidget.prototype);
+Orb.prototype.constructor = Orb;
+Orb.prototype._class += " react_Orb";
+
+Orb.prototype.mapping = mapping;
+Orb.prototype.publish("stringProp", "defaultValue", "string", "Sample Property");
+
+Orb.prototype.publish("toolbar", true, "boolean", "Show Toolbox or not", null, { tags: ["basic"] });
+Orb.prototype.publish("themeColor", "blue", "set", "Theme color", ['blue','red','black','green'], { tags: ["basic"] });
+Orb.prototype.publish("newField" ,[], "propertyArray", "Source Columns", null, { autoExpand : mapping});
```

Comment:
Yes. This is the way how dashboard designer decide which fields go to row, which go to column and which go to data. For end user, there is an option called "movable", in which dashboard designer can decide if the end user can move fields around or not. So addFields() should be kept here, right?